### DISCUSSION TITLE
Illegal access to entityID:4003! proxyID=4004   断线后reloginBaseapp失败，走login 登录流程，就会报代理错误。

### DISCUSSION THREAD

**Discussion by zsj897** [kbengine/kbengine#1349] [General]
求解决方案

**Comment by worldg** [comment]
看这个id， 应该是你account将控制权转给了avatar， login时用的不是有控制权的实体。

**Reply by zsj897** [reply]
项目里没有设计avatar，控制权一直在account

**Reply by zsj897** [reply]
如何在不关闭客户端的情况下，重启客户端KB插件。

**Comment by worldg** [comment]
走重新登录流程不应该有这个问题，  你是从loginapp登录的？ 应该会踢掉在线的账号

**Reply by zsj897** [reply]
具体流程是reloginBaseapp失败后，走登陆流程，也成功了，但只要客户端发送任何协议，就会报Illegal access to entityID 错误，踢掉客户端。

**Reply by worldg** [reply]
你搞清楚那两个实体id看看是什么实体， 再对比下客户端发的哪个实体， 服务器有控制权的是哪个实体梳理一下就知道逻辑了

**Reply by zsj897** [reply]
如果能完全看懂C++代码并有调试能力，也不会跑来这里提问了，下面的C++预编译我的能力看不明白，只能追踪在这里打印的错误日志。
 
#if defined(NETWORK_INTERFACE_DECLARE_BEGIN)
	#undef ENTITY_MESSAGE_HANDLER_STREAM
#endif

#if defined(DEFINE_IN_INTERFACE)
#if defined(BASEAPP)
#define ENTITY_MESSAGE_HANDLER_STREAM(NAME)										\
	void NAME##EntityMessagehandler_stream::handle(Network::Channel* pChannel,	\
													KBEngine::MemoryStream& s)	\
	{																			\
			ENTITY_ID eid;														\
			s >> eid;															\
																				\
			if(pChannel->isExternal() && pChannel->proxyID() != eid)			\
			{																	\
				ERROR_MSG(fmt::format(											\
					#NAME"handler::handle: Illegal access to entityID:{}! proxyID={}\n",\
					eid, pChannel->proxyID()));									\
																				\
				pChannel->condemn(#NAME"handler::handle: Illegal access");		\
																				\
				return;															\
			}

**Comment by zsj897** [comment]
已经解决了，查了一天，是因为自己的失误导致客户端实体网络包的实体id没有更新。
